### ISSUE TITLE
Tuning system glitches and shortcomings

### ISSUE THREAD

**Issue by ohlidalp** [RigsOfRods/rigs-of-rods#3122]
**Glitches:**

- [x] FlexbodyDebugUI option "hide other" doesn't work right now: https://discord.com/channels/136544456244461568/189904947649708032/1204854455737262120
- [x] Texture with the same name interferes between mods (even if they are in separate ZIPs) https://discord.com/channels/136544456244461568/189904947649708032/1204238031968673823 - fixed in 04fb5daf62870524c5b0c458e2636307292e5e87
- [x] crash when 2 addonparts tweak the same node, see below.
- [x] Flat files directly under 'mods/' cannot be loaded - https://discord.com/channels/136544456244461568/189904947649708032/1211009740688199730
- [x] Saved tuneups are not filtered for current actor if the actor has no GUID, all show up at once: https://discord.com/channels/136544456244461568/189904947649708032/1232459506714280078     - fixed by adding 'filename' directive to .tuneup file, allowing to filter by filename.

**Minor enhancements:**
- [x] add `addonpart_unwanted_exhaust` https://discord.com/channels/136544456244461568/189904947649708032/1204918739200507934 - should be minor enough
- [x] Leftover points from the original PR: https://github.com/RigsOfRods/rigs-of-rods/pull/3096#issuecomment-1839678550
- [x] Filename matching is much needed: https://discord.com/channels/136544456244461568/189904947649708032/1207163070170927195

**Comment by CuriousMike56** [comment]
[MV4S_SuspensionMods.zip](https://github.com/RigsOfRods/rigs-of-rods/files/14234833/MV4S_SuspensionMods.zip)
Apply two addons that change the same nodes -> crash
![RoR_2024-02-11_21-12-04](https://github.com/RigsOfRods/rigs-of-rods/assets/46073351/8d006b5b-a136-456e-953d-ef40cd0805f7)
![devenv_2024-02-11_21-07-10](https://github.com/RigsOfRods/rigs-of-rods/assets/46073351/0167c7f8-2061-4f6f-8a3d-ebe25cb6ae5d)

**Comment by Miner34dev** [comment]
> Ideally the game should display an error if conflicting parts (wheels, node/prop/flexbody tweaks) are installed.

I was thinking about another solution... What about addon categories? Inside the addon parts you could add some drop down menus, Which should allow to only use one of the parts in the same category at the same time.

**Comment by CuriousMike56** [comment]
In addition, the tuning menu has MP specific bugs:

- No updates to RoRNet, other players can't see applied addons. 
- Applying addons has a chance of your vehicle becoming stuck at map origin (0,0,0) for other players
- Applying any addon reloads the vehicle, the server treats this as if you spawned a new vehicle. This can trigger the spawn spam protection if enabled. 
One time the spam protection kicked me but also crashed all other players on the server. Have yet to reproduce.

Simplest workaround for now is to just disable the tuning menu when playing MP.

**Comment by Miner34dev** [comment]
> Applying any addon reloads the vehicle, the server treats this as if you spawned a new vehicle. This can trigger the spawn spam protection if enabled.
One time the spam protection kicked me but also crashed all other players on the server. Have yet to reproduce.

Can confirm, the same also happened to me.

**Comment by ohlidalp** [comment]
I looked into the crash when 2 addonparts tweak the same node. The problem is division by zero, which happens because both axis nodes of a wheel somehow end up having _precisely_ the same coordinates - as result the axis vector is <0,0,0> and you know the rest.
![obrazek](https://github.com/RigsOfRods/rigs-of-rods/assets/491088/78e214f3-cca4-4c36-b536-7e9709d76d58)
I couldn't figure out why this happens. When 2 tweaks clash on the same node, none of them actually take effect, the orignally logged tweak gets withdrawn. For example (RoR.log):
```
 ~~~ First tweak is found - added to working tuneup ~~~
 [RoR|Addonpart] INFO: file 'MV4_lowered.addonpart', element 'addonpart_tweak_node': tweaking node 119 with params { x=-1.523, y=0.266, z=-0.695 }

 ~~~ Second tweak is found - the previous one gets withdrawn ~~~
 [RoR|Addonpart] INFO: file 'MV4_rally.addonpart', element 'addonpart_tweak_node': Conflict of tweaks at node '119', addon parts '' and 'MV4_rally.addonpart'
```
I need to improve the logging - the collision message says the 2nd addonpart twice and 1st addonpart name is blank.

**Comment by ohlidalp** [comment]
@Miner34dev
> I was thinking about another solution... What about addon categories? Inside the addon parts you could add some drop down menus, Which should allow to only use one of the parts in the same category at the same time.

For the record, that makes sense but cannot be coded properly. There's no way for the game to figure out the category itself, i.e.   suspension nodes cannot be distinguished from any other node. And relying on the addonpart creator to categorize the parts correctly won't fix this issue as inevitably some modders will make mistakes.
@CuriousMike56

**Comment by ohlidalp** [comment]
Resolved by #3151
